### ISSUE TITLE
Exceptions on plugin when using MySQL (MariaDB)

### ISSUE THREAD

**Issue by brobare** [nautobot/nautobot-plugin-device-lifecycle-mgmt#28]
### Environment
* Python version:  3.9
* Nautobot version:  1.1.2
* nautobot-plugin-device-lifecycle-mgmt version: 0.2.1

### Expected Behavior
Proper access to endpoints.

### Observed Behavior
I am running nautobot with MySQL as the database backend. Plugin installation works fine. After installing the plugin, when attempting to create a Hardware LCM, I'm presented with the following error (running `debug: true`):
```
NotSupportedError at /plugins/device-lifecycle/hardware/add/
DISTINCT ON fields is not supported by this database backend
Request Method:	GET
Request URL:	https://nautobot.gcimbs.net/plugins/device-lifecycle/hardware/add/
Django Version:	3.1.13
Exception Type:	NotSupportedError
Exception Value:	
DISTINCT ON fields is not supported by this database backend
Exception Location:	/opt/nautobot/lib/python3.9/site-packages/django/db/backends/base/operations.py, line 176, in distinct_sql
Python Executable:	/usr/bin/python3.9
Python Version:	3.9.5
Python Path:	
['.',
 '',
 '/opt/nautobot',
 '/opt/nautobot/lib/python3.9/site-packages/git/ext/gitdb',
 '/opt/nautobot/bin',
 '/usr/lib/python39.zip',
 '/usr/lib/python3.9',
 '/usr/lib/python3.9/lib-dynload',
 '/opt/nautobot/lib/python3.9/site-packages',
 '/opt/nb_plugins/nautobot-routing-ospf',
 '/opt/nautobot/lib/python3.9/site-packages/gitdb/ext/smmap']
Server time:	Fri, 05 Nov 2021 08:45:39 -0800
```

### Steps to Reproduce
1. Install plugin 
2. Navigate to Hardware Notices URL - `plugins/device-lifecycle/hardware/`
3. Click on the "Add" button

**Comment by josh5276** [comment]
Hi @brobare, thanks for finding the issue! I think this is related to the way we're handling the filtering for inventory items, where we are applying a field name for distinct to apply, which is only supported on PostgreSQL.

The fix will be for us to modify the query logic in a manner that supports non-postgresql backends.

**Comment by brobare** [comment]
No worries, @josh5276! 

Figured that would be the case. Was able to run the plugin on a postgres backed dev instance and everything worked fine. Installed to our prod env (mysql) and it broke.

Really appreciate the work on this. Going to be super useful for $dayjob.

**Comment by dgarros** [comment]
Fixed in #33
